build/Shared/TaskResult.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet;
+
+#nullable enable
+
+internal static class TaskResult
```

Comment:
This is really clever, how come its not built in?

Reply:
I'm pretty sure .NET Core+ does a better job of some of this than .NET Framework.